open-ce/docker_build.py

```diff
@@ -121,37 +126,48 @@ def _stop_container(container_name):
     result = os.system(DOCKER_TOOL + " stop " + container_name)
     return result
 
-def build_in_container(image_name, output_folder, arg_strings):
+def build_in_container(image_name, args, arg_strings):
     """
     Run a build inside of a container using the provided image_name.
     """
     time_stamp = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
     container_name = IMAGE_NAME + "-" + time_stamp
 
-    _create_container(container_name, image_name, output_folder)
+    output_folder = os.path.abspath(args.output_folder)
+    env_file = os.path.abspath(args.env_config_file[0])
+    conda_build_config = os.path.abspath(args.conda_build_config)
+
+    env_folder = os.path.dirname(env_file)
+    env_file_in_container = os.path.join(HOME_PATH, "envs", os.path.basename(env_file))
```

Comment:
I think we need to do this for every file in the args.env_config_file list.

Reply:
There can't be more than one.

https://github.com/open-ce/open-ce/blob/1443d456b755c131c90892ac13c927be098aaf14/open-ce/inputs.py#L57-L63

Reply:
It used to allow more than one. Was that something that was intended?

Reply:
You can still pass multiple environment files. I think that `nargs=1` just means the number of arguments to be consumed.

The help indicates this as well, since the usage is outputted as: `env_config_file [env_config_file ...]`

Reply:
This is what I get with my current changes:
```
(dev) [bnelson@dlw05 open-ce]$ ./open-ce/open-ce build env ../open-ce-environments/envs/xgboost-env.yaml ../open-ce-environments/envs/spacy-env.yaml --docker_build --python_versions 3.7 --build_types cuda
usage: open-ce [-h] {build,test,validate} ...
open-ce: error: unrecognized arguments: ../open-ce-environments/envs/spacy-env.yaml
```

Reply:
```
(dev) [bnelson@dlw05 open-ce]$ ./open-ce/open-ce build env -h
usage: open-ce build env [-h] [--conda_build_config CONDA_BUILD_CONFIG]
                         [--output_folder OUTPUT_FOLDER]
                         [--channels CHANNELS_LIST]
                         [--repository_folder REPOSITORY_FOLDER]
                         [--python_versions PYTHON_VERSIONS]
                         [--build_types BUILD_TYPES] [--mpi_types MPI_TYPES]
                         [--cuda_versions CUDA_VERSIONS]
                         [--skip_build_packages] [--run_tests]
                         [--docker_build] [--git_location GIT_LOCATION]
                         [--git_tag_for_env GIT_TAG_FOR_ENV]
                         [--test_labels TEST_LABELS]
                         env_config_file

positional arguments:
  env_config_file       Environment config file. This should be a YAML file
                        describing the package environment you wish to build.
                        A collection of files exist under the envs directory.

optional arguments:
...
```

Reply:
I may have messed that up.
It looks like the other parts of the tools do support that being multiple files. I just never used that.

Let me look at changing it back.

Reply:
@bnemanich 
I added this back in, but it has a limitation that there will be a conflict if env files reside in different directories with the same name.
I'm looking at a way to get around that. Maybe instead of using the directory name as the mount point, I could do a concatenation of the full path without the slashes.

Reply:
I've handled this limitation. It should be good to go now.